From 46eefcaf8a53b5996fddf8f8f488beeed8984da9 Mon Sep 17 00:00:00 2001
From: duelu <luyizhou1@huawei.com>
Date: Thu, 26 Mar 2026 21:10:29 +0800
Subject: [PATCH] urma: fix default parameters of urma_perftest

the cq_mod should not be 1 when using bonding_dev with single_path enabled

Signed-off-by: duelu <luyizhou1@huawei.com>
(cherry picked from commit 0505fafaf1c99b4dadba954b23bce0e2750ecaa9)
---
 src/urma/tools/urma_perftest/perftest_parameters.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/urma/tools/urma_perftest/perftest_parameters.c b/src/urma/tools/urma_perftest/perftest_parameters.c
index 624d1fbf..32694c56 100644
--- a/src/urma/tools/urma_perftest/perftest_parameters.c
+++ b/src/urma/tools/urma_perftest/perftest_parameters.c
@@ -959,12 +959,9 @@ int check_local_cfg(perftest_config_t *cfg)
         return -1;
     }
 
-    if (strstr(cfg->dev_name, "bonding_dev") != NULL && !cfg->single_path) {
+    if (strstr(cfg->dev_name, "bonding_dev") != NULL && !cfg->single_path && !cfg->use_ctp) {
         cfg->use_ctp = true;
-        if (cfg->cq_mod != 1) {
-            cfg->cq_mod = 1;
-            (void)fprintf(stderr, "Warning, bonding_dev cq_mod should be 1.\n");
-        }
+        (void)fprintf(stderr, "Warning: ctp is enabled, and bonding_dev multi_path requires ctp to be enabled.\n");
     }
 
     if (check_time_type(cfg) != 0) {
-- 
2.43.0

